conduit/plugins/importers/filereader/fileReadWrite_test.go

```diff
@@ -0,0 +1,240 @@
+	require.NoError(t, err)
+}
+
+// numGzippedFiles returns the number of files in the importerBlockDir
+// whose filename ends in .gz
+func numGzippedFiles(t *testing.T) uint64 {
+	files, err := os.ReadDir(importerBlockDir)
+	require.NoError(t, err)
+
+	gzCount := uint64(0)
+	for _, file := range files {
+		if strings.HasSuffix(file.Name(), ".gz") {
+			gzCount++
+		}
+	}
+
+	return gzCount
+}
+
+func fileBytes(t *testing.T, path string) []byte {
+	file, err := os.Open(path)
+	require.NoError(t, err, "error opening file %s", path)
+	defer file.Close()
+
+	var buf bytes.Buffer
+	_, err = buf.ReadFrom(file)
+	require.NoError(t, err, "error reading file %s", path)
+
+	return buf.Bytes()
+}
+
+func identicalFiles(t *testing.T, path1, path2 string) {
+	var file1, file2 *os.File
+
+	defer func() {
+		if file1 != nil {
+			file1.Close()
+		}
+		if file2 != nil {
+			file2.Close()
+		}
+	}()
+
+	bytes1 := fileBytes(t, path1)
+	bytes2 := fileBytes(t, path2)
+	require.Equal(t, len(bytes1), len(bytes2), "files %s and %s have different lengths", path1, path2)
+
+	for i, b1 := range bytes1 {
+		b2 := bytes2[i]
+		require.Equal(t, b1, b2, "files %s and %s differ at byte %d (%s) v (%s)", path1, path2, i, string(b1), string(b2))
+	}
+}
+
+func uncompressBytes(t *testing.T, path string) []byte {
+	file, err := os.Open(path)
+	require.NoError(t, err, "error opening file %s", path)
+	defer file.Close()
+
+	gr, err := gzip.NewReader(file)
+	require.NoError(t, err, "error creating gzip reader for file %s", path)
+	defer gr.Close()
+
+	var buf bytes.Buffer
+	_, err = buf.ReadFrom(gr)
+	require.NoError(t, err, "error reading file %s", path)
+
+	return buf.Bytes()
+}
+func identicalFilesUncompressed(t *testing.T, path1, path2 string) {
+	var file1, file2 *os.File
+
+	defer func() {
+		if file1 != nil {
+			file1.Close()
+		}
+		if file2 != nil {
+			file2.Close()
+		}
+	}()
+
+	bytes1 := uncompressBytes(t, path1)
+	bytes2 := uncompressBytes(t, path2)
+	require.Equal(t, len(bytes1), len(bytes2), "files %s and %s have different lengths", path1, path2)
+
+	for i, b1 := range bytes1 {
+		b2 := bytes2[i]
+		require.Equal(t, b1, b2, "files %s and %s differ at byte %d (%s) v (%s)", path1, path2, i, string(b1), string(b2))
+	}
+}
+
+func getConfig(t *testing.T, pt plugins.PluginType, cfg data.NameConfigPair, dataDir string) plugins.PluginConfig {
+	configs, err := yaml.Marshal(cfg.Config)
+	require.NoError(t, err)
+
+	var config plugins.PluginConfig
+	config.Config = string(configs)
+	if dataDir != "" {
+		config.DataDir = path.Join(dataDir, fmt.Sprintf("%s_%s", pt, cfg.Name))
+		err = os.MkdirAll(config.DataDir, os.ModePerm)
+		require.NoError(t, err)
+	}
+
+	return config
+}
+
+// TestRoundTrip tests that blocks read by the filereader importer
+// under the msgp.gz encoding are written to identical files by the filewriter exporter.
+// This includes both a genesis block and a round-0 block with different encodings.
+func TestRoundTrip(t *testing.T) {
+	cleanArtifacts(t)
```

Comment:
maybe use  t.TempDir()?

Reply:
I was considering a tempdir but since I wanted to reuse the artifacts for integration testing, I found this a bit easier -as you know exactly where the artifacts are landing before the test begins. But I'm open to further discussion.

Reply:
This seems reasonable